apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetVersionDAO.java

```diff
@@ -0,0 +1,131 @@
+                change_description, metadata, created_by, last_updated_by, workspace_id
+            ) VALUES (
+                :version.id, :version.datasetId, :version.versionHash, :version.itemsCount,
+                :version.itemsAdded, :version.itemsModified, :version.itemsDeleted,
+                :version.changeDescription, :version.metadata, :version.createdBy, :version.lastUpdatedBy, :workspace_id
+            )
+            """)
+    void insert(@BindMethods("version") DatasetVersion version, @Bind("workspace_id") String workspaceId);
+
+    @SqlQuery("""
+            SELECT
+                dv.*,
+                (
+                    SELECT JSON_ARRAYAGG(tag)
+                    FROM dataset_version_tags
+                    WHERE version_id = dv.id
+                ) AS tags
+            FROM dataset_versions dv
+            WHERE dv.id = :id AND dv.workspace_id = :workspace_id
+            """)
+    Optional<DatasetVersion> findById(@Bind("id") UUID id, @Bind("workspace_id") String workspaceId);
+
+    @SqlQuery("""
+            SELECT
+                dv.*,
+                (
+                    SELECT JSON_ARRAYAGG(tag)
+                    FROM dataset_version_tags
+                    WHERE version_id = dv.id
+                ) AS tags
+            FROM dataset_versions dv
+            WHERE dv.dataset_id = :dataset_id
+                AND dv.version_hash = :version_hash
+                AND dv.workspace_id = :workspace_id
+            """)
+    Optional<DatasetVersion> findByHash(@Bind("dataset_id") UUID datasetId, @Bind("version_hash") String versionHash,
+            @Bind("workspace_id") String workspaceId);
+
+    @SqlQuery("""
+            SELECT
+                dv.*,
+                (
+                    SELECT JSON_ARRAYAGG(tag)
+                    FROM dataset_version_tags
+                    WHERE version_id = dv.id
+                ) AS tags
+            FROM dataset_versions dv
+            WHERE dv.dataset_id = :dataset_id
+                AND dv.workspace_id = :workspace_id
+            ORDER BY dv.created_at DESC
+            LIMIT :limit OFFSET :offset
+            """)
+    List<DatasetVersion> findByDatasetId(@Bind("dataset_id") UUID datasetId, @Bind("workspace_id") String workspaceId,
+            @Bind("limit") int limit, @Bind("offset") int offset);
+
+    @SqlQuery("SELECT COUNT(*) FROM dataset_versions WHERE dataset_id = :dataset_id AND workspace_id = :workspace_id")
+    long countByDatasetId(@Bind("dataset_id") UUID datasetId, @Bind("workspace_id") String workspaceId);
+
+    @SqlQuery("""
+            SELECT
+                dv.*,
+                (
+                    SELECT JSON_ARRAYAGG(tag)
+                    FROM dataset_version_tags
+                    WHERE version_id = dv.id
+                ) AS tags
+            FROM dataset_versions dv
+            WHERE dv.dataset_id = :dataset_id
+                AND dv.workspace_id = :workspace_id
+            ORDER BY dv.created_at DESC
```

Comment:
Same

Reply:
Done in https://github.com/comet-ml/opik/pull/4019/commits/69eb2525d7b89628b47aec861b6bb8ee8b997cee